cpp/src/mlt/decode/int_template.hpp

```diff
@@ -0,0 +1,226 @@
+#pragma once
+
+#include <mlt/decode/int.hpp>
+
+#include <cassert>
+#include <stdexcept>
+#include <string>
+#include <utility>
+
+namespace mlt::decoder {
+
+#pragma region Public
```

Comment:
What's this?

Reply:
Folding

![image](https://github.com/user-attachments/assets/7feb7f0c-a981-4bd1-9d51-9898a72e1edc)